client/components/suggestions/index.jsx

```diff
@@ -175,16 +186,23 @@ const Suggestions = React.createClass( {
 				continue;
 			}
 
-			filtered[ key ] = take(
-				terms[ key ].filter( term => term.indexOf( filterTerm ) !== -1 ),
-				limit
-			);
+			//check if we have show all enabled for key
```

Comment:
If we have a comment, maybe it should explain the complex part of the following code, rather than the obvious part? :)